extensions/amp-story/1.0/amp-story.js

```diff
@@ -392,6 +394,10 @@ export class AmpStory extends AMP.BaseElement {
       'en-xa',
       enXaPseudoLocaleBundle
     );
+
+    registerServiceBuilderForDoc(this.element, 'localization', function () {
+      return localizationService;
+    });
```

Comment:
Having to whitelist the usage of `registerServiceBuilderForDoc` + using a variable + using an ES5 `function` so the code can (I assume) use it as a constructor, is pretty hacky. :(

The biggest issue I see with it is that it is very prone to race conditions. Other Story components will build before the `<amp-story>` and try to access the localizationService synchronously. If this piece of code did not run before they try to access the service, everything will fail, and this execution order is not guaranteed at all.

Are you trying to register the service here, without using the `AMP.registerServiceForDoc()` helper, so we can retrieve the service synchronously and not deal with nested async code all across the Story codebase?
If so, I'm open to having the same helper we have for the store service. The first Story component trying to get the localization service would instantiate it:

https://github.com/ampproject/amphtml/blob/5b38fc6f582a16670b93a352a249b5e7d96566d0/extensions/amp-story/1.0/amp-story-store-service.js#L27-L45

Reply:
Yea, or you can use the normal pattern: have `installLocalizationService()` in `localization.js` and make sure you always call it before `Services.localizationForDoc()`.

Reply:
I actually had created a central getter in `localization.js` in a previous commit of the original PR https://github.com/ampproject/amphtml/pull/27990/commits/4c92db88577e6015d93e7af77912bcccb9d7f790, but didn't like the idea of everyone depending on `localization.js`